test/battle/ability/forecast.c

```diff
@@ -397,3 +397,21 @@ SINGLE_BATTLE_TEST("Forecast transforms Castform back when it uses a move that f
         EXPECT_EQ(GetMonData(&gPlayerParty[0], MON_DATA_SPECIES), SPECIES_CASTFORM);
     }
 }
+
+SINGLE_BATTLE_TEST("Forecast transforms Castform when Cloud Nine ability user leaves the field")
```

Comment:
I recommend adding a `PARAMETRIZE` to also handle Air Lock here

Reply:
done